src/Raven.Client/Http/DatabaseTopologyLocalCache.cs

```diff
@@ -24,80 +23,84 @@ public static void ClearTopologyFromLocalCache(string serverHash)
             catch (Exception e)
             {
                 if (_logger.IsInfoEnabled)
-                {
-                    _logger.Info("Could not clear the persisted replication information", e);
-                }
+                    _logger.Info("Could not clear the persisted database topology", e);
             }
         }
 
-        private static string GetTopologyPath(string serverHash)
+        private static string GetPath(string databaseName)
         {
-            return Path.Combine(AppContext.BaseDirectory, serverHash + ".raven-topology");
+            return Path.Combine(AppContext.BaseDirectory, databaseName + ".raven-database-topology");
         }
 
-        public static Topology TryLoadTopologyFromLocalCache(string serverHash, JsonOperationContext context)
+        public static Topology TryLoad(string databaseName, JsonOperationContext context)
```

Comment:
Database name doesn't have to be a valid path, does it?
Regardless, we also need to set the urls somehow (all of them), otherwise, we can't have two documents stores that has the same db that talk to different clusters.

Reply:
IIRC database name is validated to be a valid path.
What URLs should be used? `RequestExecutor.lastKnownUrls`?

Reply:
I think we want to use the initial urls. This way, if the user changes them, we invalid the cache (which is what I would expect).

Reply:
https://github.com/ravendb/ravendb/pull/5067